src/template-instance.ts

```diff
@@ -9,8 +10,12 @@ function* collectParts(el: DocumentFragment): Generator<TemplatePart> {
   let node
   while ((node = walker.nextNode())) {
     if (node instanceof HTMLTemplateElement) {
-      for (const part of collectParts(node.content)) {
-        yield part
+      if (node.hasAttribute('directive')) {
+        yield new InnerTemplatePart(node)
+      } else {
+        for (const part of collectParts(node.content)) {
+          yield part
+        }
```

Comment:
@keithamus Should this conditional be flattened to treat all `HTMLTemplateElement` instances as `InnerTemplatePart` instances?

That way, w could migrate the `collectParts` behavior into this package's built-in processors while still enabling consumers that provide their own custom processors with an integration seam to handle `<template>` elements however they see fit.

Reply:
Yes I agree, let's follow up separately.

Reply:
Thank you for the review! I've opened https://github.com/github/template-parts/pull/76 to remove the special-casing and make `InnerTemplatePart` available to custom processors.